File: c++/findDisappearedNumbers.cc
#include <vector>
#include <iostream>

using namespace std;

/*leetcode issue 448 */
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> ret;
        int n = nums.size();

        for(int num : nums) {
            int x = (num-1) % n;
            nums[x] += n;
        }
        for(int i = 0; i < n; ++i) {
            if(nums[i] <= n) {
                ret.push_back(i+1);

            }
        }

        return ret;
    }
};

int main() {
    Solution so;

    vector<int> ret;

    vector<int> p{4,3,2,7,8,2,3,1};
    ret = so.findDisappearedNumbers(p);
    for(int i : ret) {
        cout << i << endl;
    }
}

File: python/findDisappearedNumbers.py
#!/usr/bin/env python3
#coding=utf-8

''' leetcode issue 448'''

class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ret = []
        
        for i in nums:
            x = (i-1) % n
            nums[x] += n

        for idx, num in enumerate(nums):
            if nums[idx] <= n:
                ret.append(idx+1)
        return ret
